src/Query/Builder.php

```diff
@@ -1079,18 +1079,21 @@ protected function compileWhereBasic(array $where): array
         } // Manipulate regex operations.
         elseif (in_array($operator, ['regex', 'not regex'])) {
             // Automatically convert regular expression strings to Regex objects.
-            if (! $value instanceof Regex) {
-                $e = explode('/', $value);
-                if ($value[0] !== '/' || count($e) < 3) {
-                    throw new \InvalidArgumentException(sprintf('Regular expressions must be surrounded by slashes "/". Got "%s"', $value));
+            if (is_string($value)) {
+                $delimiter = substr($value, 0, 1);
+                if (! in_array($delimiter, ['/', '#', '~'])) {
```

Comment:
This looks like a good compromise to allow users some flexibility. Thanks :+1: 

I suggest storing this array in a constant property and then referencing it in the exception message below, though, which suggests that only "/" is permitted.